applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.cpp

```diff
@@ -0,0 +1,131 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "incremental_linear_elastic_interface_law.h"
+#include "custom_geometries/line_interface_geometry.h"
+#include "geo_mechanics_application_constants.h"
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+ConstitutiveLaw::Pointer GeoIncrementalLinearElasticInterfaceLaw::Clone() const
+{
+    return std::make_shared<GeoIncrementalLinearElasticInterfaceLaw>(*this);
+}
+
+ConstitutiveLaw::SizeType GeoIncrementalLinearElasticInterfaceLaw::WorkingSpaceDimension()
+{
+    return 2;
+}
+
+ConstitutiveLaw::SizeType GeoIncrementalLinearElasticInterfaceLaw::GetStrainSize() const
+{
+    return VOIGT_SIZE_2D_INTERFACE;
+}
+
+Vector& GeoIncrementalLinearElasticInterfaceLaw::GetValue(const Variable<Vector>& rThisVariable, Vector& rValue)
+{
+    if (rThisVariable == STRAIN) {
+        rValue = mPreviousRelativeDisplacement;
```

Comment:
Just for my understanding: this means that if we call GetValue, then initialize the material, call GetValue again, finalize the material, call GetValue again, that we get three different answers. That's probably expected, but just double checking

Reply:
So this means, that if you call CalculateMaterialResponseCauchy, where you get a certain stress and strain, and afterwards you call GetValue for the stress/strain that you get different values from these two functions (until you finalize the material)

Reply:
> Just for my understanding: this means that if we call GetValue, then initialize the material, call GetValue again, finalize the material, call GetValue again, that we get three different answers. That's probably expected, but just double checking

Yes, I believe that is correct, and also what we would expect.

Reply:
> So this means, that if you call CalculateMaterialResponseCauchy, where you get a certain stress and strain, and afterwards you call GetValue for the stress/strain that you get different values from these two functions (until you finalize the material)

Yes, that is also correct. We will only "commit" the updated values during the finalization stage.

Reply:
Clear, thanks!